app/controllers/results_submission_controller.rb

```diff
@@ -12,6 +12,28 @@ def new
     @results_validator.validate(@competition.id)
   end
 
+  def newcomer_checks
+    @competition = competition_from_params
+  end
+
+  def last_duplicate_checker_job_run
+    last_job_run = DuplicateCheckerJobRun.where(competition_id: params[:competition_id]).first
+
+    render status: :ok, json: last_job_run
+  end
+
+  def compute_potential_duplicates
+    competition = competition_from_params
```

Comment:
Here's another case where you're loading the whole competition into memory, just so that you can access its ID. Nothing else is done with this `competition` as far as I can tell.
If that's indeed the case (and I'm not missing something) then you can also refactor this to pass the `competition_id` in the job run `create!` below directly from `params`.